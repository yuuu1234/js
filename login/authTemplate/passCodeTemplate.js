import React from 'react';
import { View, ScrollView } from 'react-native';
import { observer } from 'mobx-react';
import PropTypes from 'prop-types';
import styles from './style';

const withPassCodeTemplate = ({ Header, Main, Footer }) => {
  @observer
  class Screen extends React.Component {
    handleNavigation = (screenType) => {
      const { navigation } = this.props;
      navigation.navigate(screenType);
    }

    render() {
      return (
        <View style={styles.passCodeContainer}>
          <ScrollView showsVerticalScrollIndicator={false} contentContainerStyle={{ flexGrow: 1 }}>
            { Header && <Header /> }
            { Main && <Main handleNavigation={this.handleNavigation} /> }
          </ScrollView>
          { Footer && (
            <View style={styles.passCodeFooter}>
              <Footer handleNavigation={this.handleNavigation} />
            </View>
          ) }
        </View>
      );
    }
  }

  Screen.propTypes = {
    navigation: PropTypes.shape({
      navigate: PropTypes.func.isRequired,
    }).isRequired,
  };

  return Screen;
};

export { withPassCodeTemplate };
