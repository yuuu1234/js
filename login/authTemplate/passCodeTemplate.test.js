import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import PropTypes from 'prop-types';
import { withPassCodeTemplate } from './passCodeTemplate';

const MockHeader = () => <div testId="mock-header">Header</div>;
const MockMain = ({ handleNavigation }) => <div handleNavigation={handleNavigation} testId="mock-main">Main</div>;
const MockFooter = ({ handleNavigation }) => <div handleNavigation={handleNavigation} testId="mock-footer">Footer</div>;

MockMain.propTypes = { handleNavigation: PropTypes.func.isRequired };
MockFooter.propTypes = { handleNavigation: PropTypes.func.isRequired };

describe('withPassCodeTemplate', () => {
  const params = {
    Header: MockHeader,
    Main: MockMain,
    Footer: MockFooter,
  };
  const props = {
    navigation: {
      navigate: jest.fn(),
    },
  };
  const WithAuthComponent = withPassCodeTemplate(params);
  const component = (
    <WithAuthComponent {...props} />
  );
  let wrapper;

  beforeEach(() => {
    wrapper = mount(component);
  });

  it('should render component correctly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should call navigate correctly', () => {
    const main = wrapper.find({ testId: 'mock-main' });
    main.props().handleNavigation('mock screen');

    expect(props.navigation.navigate).toHaveBeenCalledWith('mock screen');
  });
});
