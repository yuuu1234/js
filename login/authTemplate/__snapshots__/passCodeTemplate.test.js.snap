// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withPassCodeTemplate should render component correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    showsVerticalScrollIndicator={false}
  >
    <View>
      <div
        testId="mock-header"
      >
        Header
      </div>
      <div
        handleNavigation={[Function]}
        testId="mock-main"
      >
        Main
      </div>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <div
      handleNavigation={[Function]}
      testId="mock-footer"
    >
      Footer
    </div>
  </View>
</View>
`;
